### PR TITLE
new package usd

### PR DESCRIPTION
**Pull request by L-Sun**
A new package [usd](https://github.com/PixarAnimationStudios/USD)

>Universal Scene Description (USD) is an efficient, scalable system for authoring, reading, and streaming time-sampled scene description for interchange between graphics applications.

Please note that the `usd` only support its dependency `tbb` with version `2020`
https://github.com/PixarAnimationStudios/USD/issues/1471

### CODE REVIEW

File: packages/u/usd/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            "-DPXR_BUILD_IMAGING=OFF",
+            "-DPXR_BUILD_MONOLITHIC=OFF",
+            "-DPXR_BUILD_TESTS=OFF",
+            "-DPXR_BUILD_USD_IMAGING=OFF",
+            "-DPXR_ENABLE_PYTHON_SUPPORT=OFF",
+            "-DPXR_BUILD_EXAMPLES=OFF",
+            "-DPXR_BUILD_TUTORIALS=OFF",
+            "-DPXR_BUILD_USD_TOOLS=OFF",
+        }
+        if is_plat("windows") then
```
Comment (waruqi) [inline packages/u/usd/xmake.lua:31]: package:is_plat

File: packages/u/usd/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+        import("package.tools.cmake").install(package, configs)
+    end)
+
+    on_test(function (package)
+        assert(package:check_cxxsnippets({test = [[
+            void test () {
+                auto stage = pxr::UsdStage::CreateInMemory();
+            }
+        ]]}, {configs = {languages = "c++17"}, includes="pxr/usd/usd/stage.h"}))
```
Comment (waruqi) [inline packages/u/usd/xmake.lua:45]: add spaces, `includes = `

File: packages/u/usd/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            "-DPXR_BUILD_EXAMPLES=OFF",
+            "-DPXR_BUILD_TUTORIALS=OFF",
+            "-DPXR_BUILD_USD_TOOLS=OFF",
+        }
+        if is_plat("windows") then
+            table.insert(configs, "PXR_USD_WINDOWS_DLL_PATH")
+        end
+        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+        table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
+
```
Comment (waruqi) [inline packages/u/usd/xmake.lua:36]: remove empty line

File: packages/u/usd/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+             "https://github.com/PixarAnimationStudios/USD.git")
+    add_versions("v22.11", "f34826475bb9385a9e94e2fe272cc713f517b987cbea15ee6bbc6b21db19aaae")
+
+    add_deps("cmake", "boost")
+    -- usd only support tbb 2022 now https://github.com/PixarAnimationStudios/USD/issues/1471
+    add_deps("tbb 2020.3")
+
+    on_load("windows", function (package)
+        package:add("defines", "NOMINMAX")
+    end)
```
Comment (waruqi) [inline packages/u/usd/xmake.lua:15]:
```lua
if is_plat("windows") then
    add_defines("NOMINMAX")
end
```

**Comment by L-Sun** [comment]
The link errors occuring on the `window|static` may be a upstream issue

link error detials:
https://github.com/xmake-io/xmake-repo/actions/runs/3731827860/jobs/6330499532

File: packages/u/usd/xmake.lua [hunk 5]
```diff
@@ -0,0 +1,44 @@
+package("usd")
+    set_homepage("http://www.openusd.org")
+    set_description("Universal Scene Description")
+
+    add_urls("https://github.com/PixarAnimationStudios/USD/archive/refs/tags/$(version).tar.gz",
+             "https://github.com/PixarAnimationStudios/USD.git")
+    add_versions("v22.11", "f34826475bb9385a9e94e2fe272cc713f517b987cbea15ee6bbc6b21db19aaae")
+
+    add_configs("shared", {description = "Build shared library.", default = true, type = "boolean")
```
Comment (waruqi) [inline packages/u/usd/xmake.lua:9]: you can set `readonly = true` + `is_plat("windows")` to disable static

**Comment by L-Sun** [comment]
It seems that the linking error also occurs on `linux|static`. Cloud you help me find the way to solve it, or **just disable static build**?

here is infomation about linker option:
https://github.com/PixarAnimationStudios/USD/blob/release/BUILDING.md#linker-options

**Comment by waruqi** [comment]
you can check link order, and use `add_links` to adjust it. like this https://github.com/xmake-io/xmake-repo/blob/776afb4819df8c521cc807dc44407405d86bfacb/packages/k/krb5/xmake.lua#L15

**Pull request merged**